packages/fx-core/src/plugins/resource/bot/configs/scaffoldConfig.ts

```diff
@@ -48,11 +49,15 @@ export class ScaffoldConfig {
 
     this.hostType = ScaffoldConfig.getHostTypeFromProjectSettings(context);
 
-    const rawTriggers = context.answers?.[AzureSolutionQuestionNames.BotNotificationTriggers];
-    if (Array.isArray(rawTriggers)) {
-      this.triggers = rawTriggers
-        .map((rawTrigger) => utils.convertToConstValues(rawTrigger, NotificationTriggers))
-        .filter((item) => item !== undefined) as NotificationTrigger[];
+    const rawHostTypeTriggers = context.answers?.[QuestionNames.BOT_HOST_TYPE_TRIGGER];
+    if (Array.isArray(rawHostTypeTriggers)) {
+      // convert HostTypeTrigger question to trigger name
+      this.triggers = rawHostTypeTriggers
+        .map((hostTypeTrigger) => {
+          const option = HostTypeTriggerOptions.find((option) => option.id === hostTypeTrigger);
+          return option?.trigger;
+        })
+        .filter((item): item is NotificationTrigger => item !== undefined);
```

Comment:
Nit: to learn new stuff - what does `item is NotificationTrigger` mean here?

Reply:
This is called [Type Guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)
```
(item): item is NotificationTrigger => item !== undefined
```
It is to maintain type safety. With `item is NotficationTrigger`, the typescript compiler will know that "If the function returns true, then the return type of this function must be `NotificationTrigger`". So the compiler can automatically deduce the type of the whole expression is `NotificationTrigger[]`:
```
  rawHostTypeTriggers
        .map((hostTypeTrigger) => {
          const option = HostTypeTriggerOptions.find((option) => option.id === hostTypeTrigger);
          return option?.trigger;
        })
        .filter((item): item is NotificationTrigger => item !== undefined);
```

So we don't need explicit type cast (as xxx) which may break the type safety of typescript (i.e. compile time type always equals to runtime type)